code/iaas/external-handler/src/main/java/io/cattle/platform/extension/dynamic/process/ExternalHandlerActivate.java

```diff
@@ -25,42 +29,47 @@
 
     LockManager lockManager;
     GenericMapDao mapDao;
+    @Inject
+    JsonMapper jsonMapper;
 
     @Override
     public HandlerResult handle(ProcessState state, ProcessInstance process) {
         ExternalHandler externalHandler = (ExternalHandler) state.getResource();
 
-        List<String> eventNames = new ArrayList<String>();
-        DataAccessor accessor = DataAccessor.fields(externalHandler).withKey(ExternalHandlerConstants.FIELD_PROCESS_NAMES);
+        Map<String, String> processConfigs = new HashMap<String, String>();
+        DataAccessor accessor = DataAccessor.fields(externalHandler).withKey(ExternalHandlerConstants.FIELD_EXTERNAL_HANDLER_PROCESS_CONFIGS);
 
-        List<?> list = accessor.as(List.class);
+        List<? extends ExternalHandlerProcessConfig> list = accessor.asList(jsonMapper, ExternalHandlerProcessConfig.class);
         if (list != null) {
-            for (Object obj : accessor.as(List.class)) {
-                for (String part : obj.toString().trim().split("\\s*,\\s*")) {
-                    eventNames.add(part);
+            for (ExternalHandlerProcessConfig config : list) {
+                String name = config.getName();
+                for (String part : name.toString().trim().split("\\s*,\\s*")) {
```

Comment:
Looking at the PR as a whole now and how you've replaced processNames with processNameConfigs, I think you should remove this comma separation bit.

Since a process configuration is a json map and not a simple string, the split logic wont make sense.